src/apis/onepassword/OnePassword.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime, timedelta, UTC
+import json
+import logging
+from pydantic import Field
+
+from src.apis.general.Api import ApiFetcher
+from src.apis.general.PaginationSettings import PaginationSettings
+from src.apis.general.StopPaginationSettings import StopPaginationSettings
+
+logger = logging.getLogger(__name__)
+
+
+class OnePassword(ApiFetcher):
+    """
+    :param onepassword_bearer_token: The cloudflare Bearer token
+    :param pagination_off: True if pagination should be off, False otherwise
+    :param days_back_fetch: Amount of days to fetch back in the first request, Optional (adds a filter on 'start_time')
+    :param onepassword_limit: 1Password limit for number of events to return in a single request (for pagination)
+    """
+    onepassword_bearer_token: str = Field(frozen=True)
+    pagination_off: bool = Field(default=False)
+    days_back_fetch: int = Field(default=-1, frozen=True)
+    onepassword_limit: int = Field(default=100, ge=1, le=1000)
+
+    def __init__(self, **data):
+        # Initialize 1Password limit for number of events to return in a single request
+        if not data.get('onepassword_limit'):
```

Comment:
suggestion for more simple initialisation: `limit = data.pop('onepassword_limit', 100)`